src/indexing.jl

```diff
@@ -1,5 +1,5 @@
 @inline function getindex(sa::HybridArray{S}, ::Colon) where S
-    return HybridArray{S}(getindex(sa.data, :))
+    return getindex(sa.data, :)
```

Comment:
We will lose any static size information by doing this. Is there a way to prevent this? My first impression is that it might be cleaner to always return some `HybridArray` from this operation, either with `Dynamic` size (if necessary) or the statically known size (if possible). Maybe something along the lines of `hasdynamic(S)` and `tuple_nodynamic_prod(S)` could help?

Reply:
`HybridArray` without any statically known sizes doesn't provide any benefits actually, and if all dimensions are statically known then `SizedArray` from `StaticArrays` should be used (because it can take advantage of all existing `StaticArray` methods). In this case, `getindex` with `:` can result only in an array without statically known sizes (where `HybridArray` isn't helpful) or all axes of `HybridArrays` are statically sized, in which case `SizedArray` should have been used instead.

Reply:
Okay. So what about doing what you described, i.e. return `sa.data[:]` if not all sizes are known statically and a `SizedArray` otherwise?

Reply:
I'd say that the place where `HybridArray` is created with all statically known sizes should be modified in the first place but we could also have a special case for that here.

Reply:
Yes, right, that's true. I was just triggered by the failing test
```
Compatibility with EllipsisNotation: Test Failed at /home/runner/work/HybridArrays.jl/HybridArrays.jl/test/nonstandard_indices.jl:19
  Expression: typeof(u_hybrid[..]) == typeof(u_hybrid[:])
   Evaluated: HybridArray{Tuple{2,StaticArrays.Dynamic()},Float64,2,2,Array{Float64,2}} == Array{Float64,1}
```